lv_sim/sim/queue.cpp

```diff
@@ -0,0 +1,46 @@
+#include "queue.h"
+#include <stdexcept>
+#include <SDL2/SDL.h>
+
+QueueHandle_t xQueueCreate(const UBaseType_t uxQueueLength, const UBaseType_t uxItemSize)
+{
+  QueueHandle_t xQueue;
+  if (uxItemSize != 1) {
+    throw std::runtime_error("uxItemSize must be 1");
+  }
+  xQueue.queue.reserve(uxQueueLength);
+  return xQueue;
+}
+
+BaseType_t xQueueSend(QueueHandle_t &xQueue, const void * const pvItemToQueue, TickType_t xTicksToWait)
+{
+  std::lock_guard<std::mutex> guard(xQueue.mutex);
+  xQueue.queue.push_back(*reinterpret_cast<const uint8_t *const>(pvItemToQueue));
+  return true;
+}
+
+BaseType_t xQueueSendFromISR(QueueHandle_t &xQueue, const void * const pvItemToQueue, BaseType_t *xHigherPriorityTaskWoken)
+{
+  TickType_t xTicksToWait = 0;
+  xHigherPriorityTaskWoken = pdFALSE;
```

Comment:
I think this line should be `*xHigherPriorityTaskWoken = pdFALSE;`. This builds fine as it is with gcc, but with clang this throws an error.

Reply:
you're absolutely right! We should never reach that code because the isr-registers are stubbed for the `is_isr()` function in `DisplayApp.cpp` and `SystemTask.cpp` to always return `false`. But better fix it, especially when it breaks a compiler!

```suggestion
  *xHigherPriorityTaskWoken = pdFALSE;
```